api/v1alpha1/database_webhook.go

```diff
@@ -109,82 +124,91 @@ func (r *Database) Default() {
 	}
 
-	if err := util.ValidateUUID(r.Spec.NDB.ClusterId); err != nil {
-		allErrs = append(allErrs, field.Invalid(ndbPath.Child("clusterId"), r.Spec.NDB.ClusterId, "ClusterId field must be a valid UUID"))
+	if err := util.ValidateUUID(ndb.ClusterId); err != nil {
+		allErrs = append(allErrs, field.Invalid(ndbPath.Child("clusterId"), ndb.ClusterId, "ClusterId field must be a valid UUID"))
 	}
 
-	if r.Spec.NDB.CredentialSecret == "" {
-		allErrs = append(allErrs, field.Invalid(ndbPath.Child("credentialSecret"), r.Spec.NDB.CredentialSecret, "CredentialSecret must be provided"))
+	if ndb.CredentialSecret == "" {
+		allErrs = append(allErrs, field.Invalid(ndbPath.Child("credentialSecret"), ndb.CredentialSecret, "CredentialSecret must be provided in the NDB Server Spec"))
 	}
 
-	if err := util.ValidateURL(r.Spec.NDB.Server); err != nil {
-		allErrs = append(allErrs, field.Invalid(ndbPath.Child("server"), r.Spec.NDB.Server, "Server must be a valid URL"))
+	if err := util.ValidateURL(ndb.Server); err != nil {
+		allErrs = append(allErrs, field.Invalid(ndbPath.Child("server"), ndb.Server, "Server must be a valid URL"))
 	}
 
 	databaselog.Info("Exiting ndbServerSpecValidatorForCreate...")
 	return allErrs
 }
 
-func instanceSpecValidatorForCreate(r *Database, allErrs field.ErrorList, instancePath *field.Path) field.ErrorList {
+func instanceSpecValidatorForCreate(instance *Instance, allErrs field.ErrorList, instancePath *field.Path) field.ErrorList {
 	databaselog.Info("Entering instanceSpecValidatorForCreate...")
 
-	if r.Spec.Instance.DatabaseInstanceName == "" {
-		allErrs = append(allErrs, field.Invalid(instancePath.Child("databaseInstanceName"), r.Spec.Instance.DatabaseInstanceName, "A unique Database Instance Name must be specified"))
+	databaselog.Info("Logging the Instance details inside validator method", "instance", instance)
+
+	// need to assert using a regex
+	if instance.DatabaseInstanceName == nil || *instance.DatabaseInstanceName == "" {
+		allErrs = append(allErrs, field.Invalid(instancePath.Child("databaseInstanceName"), instance.DatabaseInstanceName, "A unique Database Instance Name must be specified"))
 	}
 
-	if r.Spec.Instance.Size < 10 {
-		allErrs = append(allErrs, field.Invalid(instancePath.Child("size"), r.Spec.Instance.Size, "Initial Database size must be 10 GBs or more"))
+	if instance.Size == nil || *instance.Size < 10 {
+		allErrs = append(allErrs, field.Invalid(instancePath.Child("size"), instance.Size, "Initial Database size must be specified with a value 10 GBs or more"))
 	}
 
-	if r.Spec.Instance.CredentialSecret == "" {
-		allErrs = append(allErrs, field.Invalid(instancePath.Child("credentialSecret"), r.Spec.Instance.CredentialSecret, "CredentialSecret must be provided"))
+	if instance.CredentialSecret == nil || *instance.CredentialSecret == "" {
+		allErrs = append(allErrs, field.Invalid(instancePath.Child("credentialSecret"), instance.CredentialSecret, "CredentialSecret must be provided in the Instance Spec"))
```

Comment:
If we keep the primitive types to non-pointer types then we wouldn't need to double check for nil and zero values. The zero values for these types can be validated. 
Do you know if there's any specific need to know if the user did not explicitly provide any of these?

Reply:
Since I am already going through some additional changes, I think I can also take care of this comment along with it. I am reverting it for some of the fields where it is not really necessary to convert to the pointers. That should help the code become more readable. Cc @yash-ntnx @shenoypritika @mshaaeldin